diff -ur a/libs/filesystem/src/directory.cpp b/libs/filesystem/src/directory.cpp
--- a/libs/filesystem/src/directory.cpp	2019-12-10 00:18:53.000000000 +0000
+++ b/libs/filesystem/src/directory.cpp	2020-05-23 10:08:27.731969271 +0000
@@ -54,7 +54,8 @@
   && !(defined(linux) || defined(__linux) || defined(__linux__))\
   && !defined(__ANDROID__)\
   && (!defined(__hpux) || defined(_REENTRANT)) \
-  && (!defined(_AIX) || defined(__THREAD_SAFE))
+  && (!defined(_AIX) || defined(__THREAD_SAFE))\
+  && !defined(__wasm)
 #define BOOST_FILESYSTEM_USE_READDIR_R
 #endif
 
diff -ur a/libs/filesystem/src/operations.cpp b/libs/filesystem/src/operations.cpp
--- a/libs/filesystem/src/operations.cpp	2019-12-10 00:18:53.000000000 +0000
+++ b/libs/filesystem/src/operations.cpp	2020-05-23 11:03:26.997604144 +0000
@@ -97,7 +97,8 @@
 
 #   include <sys/types.h>
 #   include <sys/stat.h>
-#   if !defined(__APPLE__) && !defined(__OpenBSD__) && !defined(__ANDROID__) \
+#   if defined(__wasm)
+#   elif !defined(__APPLE__) && !defined(__OpenBSD__) && !defined(__ANDROID__) \
  && !defined(__VXWORKS__)
 #     include <sys/statvfs.h>
 #     define BOOST_STATVFS statvfs
@@ -1036,7 +1037,9 @@
 BOOST_FILESYSTEM_DECL
 path current_path(error_code* ec)
 {
-# ifdef BOOST_POSIX_API
+# if defined(__wasm)
+  return ".";
+# elif defined(BOOST_POSIX_API)
   struct local
   {
     static bool getcwd_error(error_code* ec)
@@ -1108,7 +1111,7 @@
 BOOST_FILESYSTEM_DECL
 void current_path(const path& p, system::error_code* ec)
 {
-# ifdef UNDER_CE
+# if defined(UNDER_CE) || defined(__wasm)
   error(BOOST_ERROR_NOT_SUPPORTED, p, ec,
     "boost::filesystem::current_path");
 # else
@@ -1413,7 +1416,10 @@
   if ((prms & add_perms) && (prms & remove_perms))  // precondition failed
     return;
 
-# ifdef BOOST_POSIX_API
+# if defined(__wasm)
+  error(BOOST_ERROR_NOT_SUPPORTED, p, ec,
+    "boost::filesystem::permissions");
+# elif defined(BOOST_POSIX_API)
   error_code local_ec;
   file_status current_status((prms & symlink_perms)
                              ? fs::symlink_status(p, local_ec)
@@ -1647,9 +1653,14 @@
 BOOST_FILESYSTEM_DECL
 space_info space(const path& p, error_code* ec)
 {
-# ifdef BOOST_POSIX_API
-  struct BOOST_STATVFS vfs;
   space_info info;
+# if defined(__wasm)
+  if (!error(BOOST_ERROR_NOT_SUPPORTED,
+    p, ec, "boost::filesystem::space"))
+  {}
+
+# elif defined(BOOST_POSIX_API)
+  struct BOOST_STATVFS vfs;
   if (!error(::BOOST_STATVFS(p.c_str(), &vfs) ? BOOST_ERRNO : 0,
     p, ec, "boost::filesystem::space"))
   {
@@ -1662,10 +1673,7 @@
   }
 
 # else
-
   ULARGE_INTEGER avail, total, free;
-  space_info info;
-
   if (!error(::GetDiskFreeSpaceExW(p.c_str(), &avail, &total, &free)== 0,
      p, ec, "boost::filesystem::space"))
   {
